filebeat/crawler/registrar.go

```diff
@@ -91,7 +94,7 @@ func (r *Registrar) Run() {
 			return
 		// Treats new log files to persist with higher priority then new events
 		case state := <-r.Persist:
-			r.State[*state.Source] = state
+			r.setState(*state.Source, state)
```

Comment:
is there a potential race between prospector and harvester overwriting the state for same source?